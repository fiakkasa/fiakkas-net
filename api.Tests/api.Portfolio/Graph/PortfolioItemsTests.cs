using HotChocolate.Execution;

namespace api.Portfolio.Tests;

public class PortfolioItemsTests(GraphFixture fixture) : IClassFixture<GraphFixture>
{
    [Fact]
    public async Task PortfolioItems_Should_Return_Data()
    {
        var executor = await fixture.GetRequestExecutor();

        var result = await executor.ExecuteAsync(
"""
{
  portfolioItems(
    where: { version: { eq: 1 } }
    order: { createdAt: ASC }
  ) {
    items {
      createdAt
      href
      id
      ordinal
      title
      updatedAt
      version
      year
      category {
        createdAt
        href
        id
        title
        updatedAt
        version
      }
    }
    pageInfo {
      hasNextPage
      hasPreviousPage
    }
  }
}
""");

        Func<IQueryResult> fn = result.ExpectQueryResult;
        fn.Should().NotThrow();
        fn().Errors.Should().BeNullOrEmpty();
        result.ToJson().MatchSnapshot();
    }
}
