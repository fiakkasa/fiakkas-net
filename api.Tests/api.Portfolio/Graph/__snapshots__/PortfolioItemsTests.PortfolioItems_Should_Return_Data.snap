{
  "data": {
    "portfolioItems": {
      "items": [
        {
          "createdAt": "2024-01-01T00:00:00.000Z",
          "href": "/test",
          "id": "28e483e4-6961-4b25-88a9-d1d0a5161109",
          "ordinal": 1,
          "title": "Title",
          "updatedAt": null,
          "version": 1,
          "year": 2024,
          "category": {
            "createdAt": "2024-01-01T00:00:00.000Z",
            "href": "/test",
            "id": "38e483e4-6961-4b25-88a9-d1d0a5161109",
            "title": "Title",
            "updatedAt": null,
            "version": 1
          }
        }
      ],
      "pageInfo": {
        "hasNextPage": false,
        "hasPreviousPage": false
      }
    }
  }
}
